stubs/sklearn/utils/_metadata_requests.pyi

```diff
@@ -70,6 +70,15 @@ class RequestMethod:
     def __get__(self, instance, owner): ...
 
 class _MetadataRequester:
+    def __init_subclass__(cls, **kwargs): ...
+    def get_metadata_routing(self): ...
+    # This code is never run in runtime, but it's here for type checking.
+    # Type checkers fail to understand that the `set_{method}_request`
+    # methods are dynamically generated, and they complain that they are
+    # not defined. We define them here to make type checkers happy.
+    # During type checking analyzers assume this to be True.
+    # The following list of defined methods mirrors the list of methods
+    # in SIMPLE_METHODS.
```

Comment:
I just copied the message from the source here to explain why these methods are there despite not existing at runtime.